### PR TITLE
Walletcmd wip

### LINKED ISSUE #1429: Change Public Passphrase
I would like to request the ability to change the public passphrase of wallet.db. Right now, the only command available is dcrctl --wallet walletpassphrasechange "old" "new" but this leaves the public pass unaffected.

### PR DESCRIPTION
**Pull request by Kifen**
New feature to enable the change of a public pass phrase over Json-rpc. It resolves #1429.

### CODE REVIEW

File: p2p/go.mod [hunk 1]
```diff
@@ -20,4 +20,4 @@ require (
 	golang.org/x/sys v0.0.0-20181212120007-b05ddf57801d // indirect
 	gopkg.in/fsnotify.v1 v1.4.7 // indirect
 	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
-)
+)
```
Comment (itswisdomagain) [inline p2p/go.mod]: Undo.

File: rpc/walletrpc/go.mod [hunk 2]
```diff
@@ -3,4 +3,4 @@ module github.com/decred/dcrwallet/rpc/walletrpc
 require (
 	github.com/golang/protobuf v1.3.2
 	google.golang.org/grpc v1.22.0
-)
+)
```
Comment (itswisdomagain) [inline rpc/walletrpc/go.mod]: Undo.

**Review by itswisdomagain — CHANGES_REQUESTED** [review]

File: internal/rpc/jsonrpc/methods.go [hunk 3]
```diff
@@ -133,7 +133,7 @@ var handlers = map[string]handler{
 	"walletlock":              {fn: (*Server).walletLock},
 	"walletpassphrase":        {fn: (*Server).walletPassphrase},
 	"walletpassphrasechange":  {fn: (*Server).walletPassphraseChange},
-
+	"walletpubpassphrasechange": {fn: (*Server).walletPubPassphraseChange},
```
Comment (itswisdomagain) [inline internal/rpc/jsonrpc/methods.go:136]: There was an empty line space here that demarcated 2 sections of code. Restore the empty line space after the `walletpubpassphrasechange` line.

File: internal/rpchelp/helpdescs_en_US.go [hunk 4]
```diff
@@ -559,6 +559,11 @@ var helpDescsEnUS = map[string]string{
 	"walletpassphrase-passphrase": "The wallet passphrase",
 	"walletpassphrase-timeout":    "The number of seconds to wait before the wallet automatically locks",
 
+	//WalletPubPassPhraseChangeCmd help
```
Comment (itswisdomagain) [inline internal/rpchelp/helpdescs_en_US.go:562]: There should be a single whitespace after `//`. Compare other help comments.

File: rpc/jsonrpc/types/methods.go [hunk 5]
```diff
@@ -1281,6 +1281,21 @@ type MixOutputCmd struct {
...
 
+// WalletPubPassphraseChangeCmd defines the walletpubpassphrasechange JSON-RPC command.
+type WalletPubPassphraseChangeCmd struct {
+	OldPassphrase string
+	NewPassphrase string
+}
+
+// NewWalletPubPassphraseChangeCmd returns a new instance which can be used to
+// issue a walletpubpassphrasechange JSON-RPC command.
+func NewWalletPubPassphraseChangeCmd(oldPassphrase, newPassphrase string) *WalletPubPassphraseChangeCmd {
```
Comment (itswisdomagain) [inline rpc/jsonrpc/types/methods.go:1292]: iirc, jrick said he doesn't want this. refer https://github.com/decred/dcrwallet/pull/1510#discussion_r316240839. 

I'm not sure this function is called from anywhere either, so you can just delete it.

**Review by itswisdomagain — CHANGES_REQUESTED** [review]

File: internal/rpc/jsonrpc/methods.go (hunk 3 above)
Comment (Kifen) [reply internal/rpc/jsonrpc/methods.go:136]: Restored the empty line space demarcating sections of code

**Pull request closed without merging**